modules/exploits/linux/http/ipfire_oinkcode_exec.rb

```diff
@@ -66,31 +66,35 @@ def check
         'method'        => 'GET',
         'authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
       )
-      fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
-      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") if res.code != 200
-      /\<strong\>IPFire (?<version>[\d.]{4}) \([\w]+\) - Core Update (?<update>[\d]+)/ =~ res.body
 
-      if version && update && version.eql? "2.19" && update.to_i <= 110
+      if res and res.code == 200
+        /\<strong\>IPFire (?<version>[\d.]{4}) \([\w]+\) - Core Update (?<update>[\d]+)/ =~ res.body
+      end
+
+      # now that we've pulled the info we need, check version.
+      if version && update && version.eql == '2.19' && update.to_i <= 110
```

Comment:
`version.eql == '2.19'` is incorrect syntax. It should be `version == '2.19'` or `version.eql? '2.19'`